unified-runtime/source/adapters/cuda/device.hpp

```diff
@@ -102,11 +104,28 @@ struct ur_device_handle_t_ {
     if (MemoryProviderShared) {
       umfMemoryProviderDestroy(MemoryProviderShared);
     }
+    if (NVMLUsed) {
+      UR_CHECK_ERROR(nvmlShutdown());
+    }
     cuDevicePrimaryCtxRelease(CuDevice);
   }
 
   native_type get() const noexcept { return CuDevice; };
 
+  nvmlDevice_t getNVML() {
+    // Initialization happens lazily once per device object. Call to nvmlInit by
+    // different objects will just increase the reference count. Each object's
+    // destructor calls shutdown method, so once there will be no NVML users
+    // left, resources will be released.
+    std::call_once(NVMLInitFlag, [this]() {
+      UR_CHECK_ERROR(nvmlInit());
+      NVMLUsed = true;
+    });
+    nvmlDevice_t NVMLDevice;
```

Comment:
[NIT]: I think we could define `NVMLDevice` before `call_once` and call `nvmlDeviceGetHandleByIndex` inside `call_once`. This will save us some overhead of calling `nvmlDeviceGetHandleByIndex` every time `getNVML` is used.

Reply:
Done, thanks.